src/components/Map/index.tsx

```diff
@@ -1,18 +1,62 @@
-import Map from "react-map-gl/maplibre";
+"use client";
 import mapStyle from "./mapStyle";
-import { FC } from "react";
+import { FC, useRef } from "react";
+import maplibregl from "maplibre-gl";
+import Map, { Source, Layer, MapLayerMouseEvent } from "react-map-gl/maplibre";
+import { layerStyles } from "./layerStyles";
 
-interface Props {}
-export const MapComponent: FC<Props> = () => {
+interface Props {
+  mapData: any;
+  activeDistrictSet: (label: string) => void;
+}
+
+const districtIDs = [
+  "district_1",
+  "district_2",
+  "district_3",
+  "district_4",
+  "district_5",
+];
+
+export const MapComponent: FC<Props> = ({ mapData, activeDistrictSet }) => {
+  const mapRef = useRef<mapboxgl.Map>();
+  const onMapCLick = (event: MapLayerMouseEvent) => {
+    event.preventDefault();
+    if (!mapRef || !mapRef.current) {
```

Comment:
A `ref` should always exist, even if it doesn’t have any "content" in `current`. So, you can skip the first check! #leancode

```suggestion
    if (!mapRef.current) {
```

Reply:
check!